docs/reference/template-module.md

```diff
@@ -49,11 +49,20 @@ The important keys that a module has are listed below, but an exhaustive list ca
 - `name` - The import path of the module
 - `description` - A description of the module
 - `modules` - a list of modules that this module depends on
-  _ `name` - import path of the module depended on
-  _ `version` - optional: A version to pin this module to.
+  - `name` - import path of the module depended on
+  - `version` - optional: A version to pin this module to.
+- `dirReplacements` - a key:value mapping of template-able replacements for directory names, often used for languages like Java/Kotlin with directories named after the projects
+  - key: The directory name to replace
+  - value: The template-able replacement name
+  - example: This k:v pair will take the `com.projname` directory and replace it with the result of rendering the template (replacing it with the contents of the module's argument named "project-name"):
+
+```
+  "src/main/kotlin/com.projname": '{{ stencil.Arg "project-name" }}'
+```
```

Comment:
Does it replace `com.projname` _literally_, or the entire string?

Reply:
just the folder name, under that path -- i don't want to enable directory structure remaps with this

Reply:
added docs and tests against this assertion